src/testRunner/unittests/evaluation/awaitUsingDeclarations.ts

```diff
@@ -0,0 +1,1614 @@
+                output.push(e);
+            }
+            output.push("after try");
+        }
+        `, { target: ts.ScriptTarget.ES2018 }, { SuppressedError: FakeSuppressedError });
+
+        await main();
+
+        assert.deepEqual(output, [
+            "before try",
+            "enter try",
+            "body",
+            "disposed",
+            {
+                error: "dispose error",
+                suppressed: "body error"
+            },
+            "after try"
+        ]);
+    });
+
+    it("'await using' in Block, 'return' in body (es2018)", async () => {
+        const { main, output } = evaluator.evaluateTypeScript(`
+        export const output: any[] = [];
+
+        const disposable = {
+            async [Symbol.asyncDispose]() {
+                output.push("disposed");
+            }
+        };
+
+        function body() {
+            output.push("body");
+        }
+
+        export async function main() {
+            output.push("before block");
+            {
+                output.push("enter block");
+                await using _ = disposable;
+                body();
+                return;
+            }
+            output.push("after block");
+        }
+        `, { target: ts.ScriptTarget.ES2018 });
+
+        await main();
+
+        assert.deepEqual(output, [
+            "before block",
+            "enter block",
+            "body",
+            "disposed"
+        ]);
+    });
+
+    it("'await using' in Block, 'break' in body (es2018)", async () => {
+        const { main, output } = evaluator.evaluateTypeScript(`
+        export const output: any[] = [];
+
+        const disposable = {
+            async [Symbol.asyncDispose]() {
+                output.push("disposed");
+            }
+        };
+
+        function body() {
+            output.push("body");
+        }
+
+        export async function main() {
+            output.push("before block");
+            for (let i = 0; i < 2; i++) {
+                output.push("enter block");
+                await using _ = disposable;
+                body();
+                break;
+            }
+            output.push("after block");
+        }
+        `, { target: ts.ScriptTarget.ES2018 });
+
+        await main();
+
+        assert.deepEqual(output, [
+            "before block",
+            "enter block",
+            "body",
+            "disposed",
+            "after block"
+        ]);
+    });
+
+    it("'await using' in Block, 'continue' in body (es2018)", async () => {
+        const { main, output } = evaluator.evaluateTypeScript(`
+        export const output: any[] = [];
+
+        const disposable = {
+            async [Symbol.asyncDispose]() {
+                output.push("disposed");
+            }
+        };
+
+        function body() {
+            output.push("body");
+        }
+
+        export async function main() {
+            output.push("before block");
+            for (let i = 0; i < 1; i++) {

```

Comment:
it would be nice to see two iterations instead of one

Reply:
Can you clarify what you mean? Are you talking about two "parallel"/interleaved loops?